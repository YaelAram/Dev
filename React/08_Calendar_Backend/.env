PORT=8080
PRIVATE_KEY=BlueCoffeeBlackie
DB_CNN=mongodb+srv://NodeUser:hpaHnnprt3v06bqI@nodetest.hxx05.mongodb.net/mern_calendar
