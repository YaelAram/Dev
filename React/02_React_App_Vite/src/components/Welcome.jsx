import React from 'react';
import PropTypes from 'prop-types';

function Welcome( { title, subtitle, message } ) {
    return (
        <React.Fragment>
            <h1>{ title }</h1>
            <h2>{ subtitle }</h2>
            <p>{ message }</p>
        </React.Fragment>
    );
};

Welcome.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string.isRequired,
    message: PropTypes.number.isRequired
};

export default Welcome;
