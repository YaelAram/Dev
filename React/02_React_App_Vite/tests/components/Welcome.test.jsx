import React from 'react';
import { render, screen } from "@testing-library/react";
import Welcome from "../../src/components/Welcome";

describe( 'Welcome component', () => {
    const title = 'Hello World';
    const subtitle = 'Goodbye World';
    const message = 123;

    test( '<Welcome /> Snapshot', () => {
        const { container } = render( <Welcome title={ title } subtitle={ subtitle } message={ message }/> );
        expect( container ).toMatchSnapshot();
    } );

    test( '<Welcome /> Contain H1 with title', () => {
        // To watch what is inside screeen use screen.debug()
        render( <Welcome title={ title } subtitle={ subtitle } message={ message }/> );
        expect( screen.getByRole( 'heading', { level: 1 } ).innerHTML ).toContain( title );
    } );

    test( '<Welcome /> Contain H2 with subtitle', () => {
        // To check for 2 or more apperances of the text user getAllByText it returns an array
        render( <Welcome title={ title } subtitle={ subtitle } message={ message }/> );
        expect( screen.getByRole( 'heading', { level: 2 } ).innerHTML ).toContain( subtitle );
    } );

    test( '<Welcome /> Contain p tag with message', () => {
        // To check for 2 or more apperances of the text user getAllByText it returns an array
        render( <Welcome title={ title } subtitle={ subtitle } message={ message }/> );
        expect( screen.getByText( message ).innerHTML ).toContain( String( message ) );
    } );
} );
