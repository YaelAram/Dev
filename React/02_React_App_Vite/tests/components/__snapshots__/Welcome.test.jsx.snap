// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome component <Welcome /> Snapshot 1`] = `
<div>
  <h1>
    Hello World
  </h1>
  <h2>
    Goodbye World
  </h2>
  <p>
    123
  </p>
</div>
`;
